migrate/migrations/20190701130233-verification-codes-update.sql

```diff
@@ -0,0 +1,54 @@
+-- +migrate Up
+UPDATE engine_processing_versions
+SET "version" = 2
+WHERE type_id = 'verify';
+
+-- add new columns
+ALTER TABLE user_verification_codes
+    ADD COLUMN contact_method_id UUID REFERENCES user_contact_methods (id) UNIQUE,
```

Comment:
If I recall correctly @mastercactapus and I didn't add `CASCADE` as it could potentially delete outgoing messages as well, if any existed

Reply:
If i'm not wrong, `ON DELETE CASCADE` on contact_method_id column means that if the foreign table reference (user_contact_methods (id)) is deleted, it will also delete the column from user_verification_codes too. I am not sure how that would delete outgoing_messages. Is there a trigger somewhere that does this? 
Also, if the contact method itself is deleted, we do want the corresponding outgoing_messages to be deleted too.
What do you think?

Reply:
Whoops, I was thinking of something else 😅

Reply:
The not wanting to delete the row here was specifically during the initial migration so that we don't drop in-flight messages.

Deleting a contact method, however, would be an appropriate reason to cleanup verification codes. Consequently deleting the CM will cleanup everything, including `outgoing_messages` for that CM.

Good catch